### PR TITLE
MinimalDataParser add options, allow Nerfstudio data format to specify test indices

### PR DESCRIPTION
**Pull request by jkulhanek**
PR allows more options for the MinimalDataparser and allows NerfstudioDataparser to specify train/test indices in the json.

### CODE REVIEW

**Comment by jkulhanek** [comment]
It also tracks the changes to the coord system done in the ns-process-data.

**Comment by jkulhanek** [comment]
I am unsure if we should track the "applied_transform" done by ns-process data. This would allow us to get back the original COLMAP poses if needed.

File: nerfstudio/data/dataparsers/nerfstudio_dataparser.py [hunk 1]
```diff
@@ -182,22 +182,30 @@ def _generate_dataparser_outputs(self, split="train"):
...
-            indices = i_train
-        elif split in ["val", "test"]:
-            indices = i_eval
+        if f"{split}_indices" in meta:
+            # Datasets sets the train/test indices
+            indices = meta[f"{split}_indices"]
+            CONSOLE.log(f"[yellow] Dataset is overriding {split}_indices to {indices}")
+            indices = np.array(indices, dtype=np.int32)
+            if indices.max() >= len(poses):
+                raise RuntimeError(f"Invalid dataset indices: index {indices.max()} > {len(poses)} dataset images.")
```
Comment (tancik) [inline nerfstudio/data/dataparsers/nerfstudio_dataparser.py:191]: Instead of storing indicies, can we store filenames?

**Comment by jkulhanek** [comment]
I have also added some simple tests for nerfstudio_parser. It can be extended in future.

**Review by tancik — APPROVED** [review]
LGTM, thanks for making the changes to use filenames.

**Pull request merged**